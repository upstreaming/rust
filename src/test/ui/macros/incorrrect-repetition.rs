macro_rules! sneaky {
    ($($i:ident $e:expr)*) => {}
    //~^ ERROR `$e:expr` is followed (through repetition) by `$i:ident`, which is not allowed for
}

fn main() {
    sneaky!(a b c d);
}
