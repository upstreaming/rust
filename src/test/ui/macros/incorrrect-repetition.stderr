error: `$e:expr` is followed (through repetition) by `$i:ident`, which is not allowed for `expr` fragments
  --> $DIR/incorrrect-repetition.rs:2:17
   |
LL |     ($($i:ident $e:expr)*) => {}
   |        -------- ^^^^^^^ this fragment is followed by the first fragment in this repetition without a valid separator
   |        |
   |        this is the first fragment in the evaluated repetition

error: aborting due to previous error

